lab_1_classify_by_unigrams/main.py

```diff
@@ -135,6 +233,29 @@ def preprocess_profile(profile: dict) -> dict[str, str | dict] | None:
     In case of corrupt input arguments or lack of keys 'name', 'n_words' and
     'freq' in arguments, None is returned
     """
+    if not isinstance(profile, dict):
+        return None
+    if (len(profile.keys()) != 3
+            or not all(x in profile for x in ['freq', 'name', 'n_words'])):
+        return None
+    processed_profile = {'name': profile['name'], 'freq': {}}
+    dictionary = {}
+    for unigram in profile['freq'].keys():
+        if unigram.isalpha():
+            pass
+        if len(unigram) == 1:
+            dictionary.setdefault(unigram, profile['freq'][unigram])
+    letters = list(dictionary)
+    for letter in letters:
+        if not letter.isupper():
```

Comment:
why isupper()

Reply:
because some letters do not have a capital letter, for example, hieroglyphs (there was one). it is better to consider the case when the letter is not upper than when it is lower. moreover, then I check only the upper ones and fix them too